applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/harmonic_analysis_strategy.hpp

```diff
@@ -666,14 +666,24 @@ class HarmonicAnalysisStrategy
             {
                 if( !it_dof->IsFixed() )
                 {
-                    //absolute displacement
-                    it_dof->GetSolutionStepValue(step) = std::abs(rModalDisplacement(it_dof->EquationId()));
+                    const auto modal_displacement = rModalDisplacement( it_dof->EquationId() );
+                    //displacement
+                    if( std::real( modal_displacement ) < 0 )
```

Comment:
@loumalouomega This seems cumbersome but I don't know a better way, do you have a suggestion?

Reply:
ping @loumalouomega